### PR TITLE
[ticket/13348] Fetch complete rowset and return data row by row

### PR DESCRIPTION
**Pull request by marc1706**
Instead of keeping $result as a static pointer and calling sql_fetchrow()
on every iteration of get_time(), fetch the complete set ofrows upon querying
the database. Afterwards, return the rows one by one using array_shift().

Ticket: https://tracker.phpbb.com/browse/PHPBB3-13348

### CODE REVIEW

File: phpBB/phpbb/feed/base.php [hunk 1]
```diff
@@ -229,9 +229,9 @@ function get_passworded_forums()
 
 	function get_item()
 	{
-		static $result;
+		static $feed_row;
```
Comment (nickvergessen) [inline phpBB/phpbb/feed/base.php]: use a member instead of a static variable for better testability?

File: phpBB/phpbb/feed/base.php [hunk 2]
```diff
@@ -229,9 +229,9 @@ function get_passworded_forums()
...
 
-		if (!isset($result))
+		if (!isset($feed_row))
```
Comment (nickvergessen) [inline phpBB/phpbb/feed/base.php]: then check `!is_array()` here

Comment (marc1706) [inline phpBB/phpbb/feed/base.php]: You mean (!isset($feed_row) || !is_array($feed_row)) ?

Comment (marc1706) [inline phpBB/phpbb/feed/base.php]: Oh, you meant just if it's an array

File: phpBB/phpbb/feed/base.php [hunk 3]
```diff
@@ -67,6 +67,9 @@
 	*/
 	var $separator_stats = "\xE2\x80\x94"; // &mdash;
 
+	/** @var array Feed database rows */
+	protected $feed_row;
```
Comment (nickvergessen) [inline phpBB/phpbb/feed/base.php]: rows?

**Comment by bantu** [comment]
Why is https://github.com/marc1706/phpbb/blob/ticket/13348/phpBB/phpbb/feed/base.php#L125-134 insufficient?

**Comment by marc1706** [comment]
I guess because it wasn't used at all yet? I guess because the class was called statically. Anyhow, didn't see this but it looks like we should be able to use that then instead.

**Comment by bantu** [comment]
@marc1706 Use here. https://github.com/phpbb/phpbb/blob/develop/phpBB/feed.php#L151

**Comment by marc1706** [comment]
@bantu I meant that $this->result wasn't used. Therefore calling the close() method didn't do anything.

**Comment by bantu** [comment]
@marc1706 Ah. Ok. Very good. Is olympus affected too?

**Comment by bantu** [comment]
It is.

**Comment by marc1706** [comment]
Yeah, the olympus PR probably won't merge into dev-asc though. I'll create a new PR for olympus right away though.

**Comment by marc1706** [comment]
Replacement PR against develop-olympus: https://github.com/phpbb/phpbb/pull/3365

**Pull request closed without merging**